### PR TITLE
PosixTimevalTest - Add test cases to increase code coverage to 100% f…

### PR DESCRIPTION
**Pull request by chmorgan**
…or PosixTimeval

### CODE REVIEW

**Comment by chmorgan** [comment]
Note that codecov appears to be wrong. After going through the code paths a few times and the tests we should be at 100% coverage here but it only shows 92%. Maybe it will go up when it runs on master or something.

File: Test/PosixTimevalTest.cs [hunk 1]
```diff
@@ -60,5 +73,24 @@ public void ToStringTest()
...
+        public void HashTest()
+        {
+            Assert.AreNotEqual(p1.GetHashCode(), p2.GetHashCode());
+        }
+
+        [Test]
+        public void DateTimeConversion()
+        {
+            var now = System.DateTime.Now;
+            var pX = new PosixTimeval(now);
```
Comment (kayoub5) [inline Test/PosixTimevalTest.cs:87]: `Assert.AreEqual(pX.Date, now)`

File: Test/PosixTimevalTest.cs [hunk 2]
```diff
@@ -60,5 +73,24 @@ public void ToStringTest()
...
+        }
+
+        [Test]
+        public void EmptyConstructor()
+        {
+            var pX = new PosixTimeval();
```
Comment (kayoub5) [inline Test/PosixTimevalTest.cs:93]: `Assert.AreEqual(pX.Date, DateTime.UtcNow)`

File: Test/PosixTimevalTest.cs [hunk 3]
```diff
@@ -60,5 +73,24 @@ public void ToStringTest()
 
             Assert.AreEqual("123.012345s", p1.ToString());
         }
+
+        [Test]
+        public void HashTest()
+        {
+            Assert.AreNotEqual(p1.GetHashCode(), p2.GetHashCode());
```
Comment (kayoub5) [inline Test/PosixTimevalTest.cs:80]: `Assert.NotEqual(p1.GetHashCode(), p4.GetHashCode());`

**Comment by kayoub5** [comment]
@chmorgan current `PosixTimeval` supports only microsecond precision, recent versions of libpcap provides nanosecond precision.

**Comment by chmorgan** [comment]
@kayoub5 good catch on nanosecond precision. I'll look at adding that support.

**Pull request merged**